### ISSUE TITLE
Block style variations

### ISSUE THREAD

**Issue by benoitchantre** [wprig/wprig#257]
## Issue Overview

There are good chances that a theme register block style variations.

Is that something we want in WP Rig? If yes, how (code or documentation)?

**Comment by benoitchantre** [comment]
I think this would be added to the Editor component.

**Comment by samikeijonen** [comment]
If needed this can be using [block filters](https://wordpress.org/gutenberg/handbook/designers-developers/developers/filters/block-filters/).

**Comment by mor10** [comment]
Yes to the idea. We should add an example as an optional component, including both adding style variations and hiding blocks using block filters as @samikeijonen pointed out. Target 2.1 or thereabouts.

**Comment by benoitchantre** [comment]
If we allow us to hide some core styles, I think the Button block is a good candidate for that.

I don’t understand why someone can choose between square or rounded buttons. This decision is made by the designer, not by the editor and once decided all buttons instances should follow the same style. Eventually, a style variation could be used to introduce a secondary button style if we think it’s a good idea to have primary and secondary styles.

**Comment by benoitchantre** [comment]
> We should add an example as an optional component

I’m not sure to understand why we would need a new component. I think the _Editor_ component would be approrpriate to enqueue the required script(s) to add/remove block variations. In PHP, we could comment the action used to enqueue the script(s).

I’ll work on a PR to see how it can work.

**Comment by benoitchantre** [comment]
WordPress 5.2 should include a UI to disable blocks. I don’t think this is still relevant to provide an example for that.

I was thinking to add a variation for the list style. Something different than bullets, maybe checkmarks. Ideas are welcome.

A pull (draft) pull request should land soon.
